jetson/teleop/src/__main__.py

```diff
@@ -86,66 +87,68 @@ def deadzone(magnitude, threshold):
     else:
         temp_mag = (temp_mag - threshold)/(1 - threshold)
 
-    return math.copysign(temp_mag, magnitude)
+    return copysign(temp_mag, magnitude)
 
 
-def joystick_math(new_motor, magnitude, theta):
-    new_motor.left = abs(magnitude)
-    new_motor.right = new_motor.left
+class Drive:
+    def __init__(self, reverse: bool):
+        self.auton_enabled = False
+        self.reverse = reverse
 
-    if theta > 0:
-        new_motor.right *= 1 - (theta * 0.75)
-    elif theta < 0:
-        new_motor.left *= 1 + (theta * 0.75)
+    def reverse_callback(self, channel, msg):
+        self.reverse = ReverseDrive.decode(msg).reverse
 
-    if magnitude < 0:
-        new_motor.left *= -1
-        new_motor.right *= -1
-    elif magnitude == 0:
-        new_motor.left += theta
-        new_motor.right -= theta
-
-
-def drive_control_callback(channel, msg):
-    global kill_motor, connection
-
-    if not connection:
-        return
-
-    input_data = Joystick.decode(msg)
-
-    if input_data.kill:
-        kill_motor = True
-    elif input_data.restart:
-        kill_motor = False
-
-    if kill_motor:
-        send_drive_kill()
-    else:
-        new_motor = DriveVelCmd()
-        input_data.forward_back = -quadratic(input_data.forward_back)
-        magnitude = deadzone(input_data.forward_back, 0.04)
-        theta = deadzone(input_data.left_right, 0.1)
-
-        joystick_math(new_motor, magnitude, theta)
-
-        damp = (input_data.dampen - 1)/(2)
-        new_motor.left *= damp
-        new_motor.right *= damp
-
-        lcm_.publish('/drive_vel_cmd', new_motor.encode())
+    def auton_enabled_callback(self, channel, msg):
+        self.auton_enabled = AutonState.decode(msg).is_auton
 
+    def teleop_drive_callback(self, channel, msg):
+        if self.auton_enabled:
+            return
 
-def autonomous_callback(channel, msg):
-    input_data = Joystick.decode(msg)
-    drive_command = DriveVelCmd()
+        input = Joystick.decode(msg)
+
+        # TODO: possibly add quadratic control
+        linear = deadzone(input.forward_back, 0.05) * input.dampen
+        angular = deadzone(input.left_right, 0.1) * input.dampen
+
+        # Convert arcade drive to tank drive
+        # TODO: possibly flip signs of rotation if necessary
+        angular_op = (angular / 2) / (abs(linear) + 0.5)
+        vel_left = linear + angular_op
+        vel_right = linear - angular_op
+
+        # Account for reverse
+        if self.reverse:
+            tmp = vel_left

```

Comment:
(for in the future) i dont think this produces a true polarity flip, like it flips turn + driving at the same time bc they're both positive i think, but for turning in place like where left turn is like L=-1, R=1, like this 'flips' it to L=-1 and R=1 which when the polarity is flipped is a right turn?

Reply:
I'm not following. Turning left in place is the same for both directions -- it's turning counterclockwise.

Reply:
but if you're flipped that's a right turn

Reply:
we can talk about it offline since it's not critical to this PR, but i think this explains why in testing we noticed even from the joystick curve turning to the left one way would make the rover go left but a left turn in place would make it go right